src/XRP/model/xrp-payment.ts

```diff
@@ -31,6 +36,16 @@ export default class XRPPayment {
 
     const destinationTag = payment.getDestinationTag()?.getValue()
 
+    const destinationXAddress = Utils.encodeXAddress(
+      destination,
+      destinationTag,
+      xrplNetwork === XRPLNetwork.Test,
+    )
+    // An X-address should always be able to be encoded.
+    if (!destinationXAddress) {
+      return undefined
+    }
```

Comment:
Instead of returning `undefined`, should we just use a `!` not-null operator:
```ts
// eslint-disable-next-line ...
const xAddress = Utils.encodeXAddress(...)!
```

Reply:
So I'm inclined to continue returning `undefined` here because that is how we're treating other error states that should never happen in all of these wrapper classes.  The X-address *should* always be encodable, but if it isn't, I do want the whole object to be rotten.

Reply:
+1 to Amie's point here - we shouldn't use force unwraps in non-test code IMO. 

I realize this is a logical fallacy right now, but there could be **some** point in the future where preconditions change and this is no longer true. Then force unwraps will fail and things will break in unexpected ways at runtime